unittests/Interpreter/InterpreterInfra.cpp

```diff
@@ -223,6 +247,62 @@ bool KBTesting::nextKB(llvm::KnownBits &x) {
   return false;
 }
 
+bool testKB(llvm::KnownBits Calculated, EvalValueKB Expected,
+            Inst::Kind K, std::vector<llvm::KnownBits> KBS) {
+    // expected value is poison/ub; so let binary transfer functions do
+  // whatever they want without complaining
+  if (!Expected.hasValue())
+    return true;
+
+  if (Calculated.getBitWidth() != Expected.ValueKB.getBitWidth()) {
+    llvm::errs() << "Expected and Given have unequal bitwidths - Expected: "
+                  << Expected.ValueKB.getBitWidth() << ", Given: " << Calculated.getBitWidth() << '\n';
+    return false;
+  }
+  if (Calculated.hasConflict() || Expected.ValueKB.hasConflict()) {
+    llvm::errs() << "Expected or Given result has a conflict\n";
+    return false;
+  }
+
+  if (KnownBitsAnalysis::isConflictingKB(Calculated, Expected.ValueKB)) {
+    outs() << "Unsound!! " << Inst::getKindName(K) << "\nInputs: ";
+    for (auto KB : KBS) {
+      outs() << KnownBitsAnalysis::knownBitsString(KB) << " ";
+    }
+    outs() << "\nCalculated: " << KnownBitsAnalysis::knownBitsString(Calculated) << '\n';
+    outs() << "Expected: " << KnownBitsAnalysis::knownBitsString(Expected.ValueKB) << '\n';
+    return false;
+  }
+  return true;
+}
+
+bool KBTesting::testTernaryFn(Inst::Kind K, size_t Op0W,
+                              size_t Op1W, size_t Op2W) {
+  llvm::KnownBits x(Op0W);
+  do {
+    llvm::KnownBits y(Op1W);
+    do {
+      llvm::KnownBits z(Op2W);
+      do {
+        InstContext IC;
+        auto Op0 = IC.getInst(Inst::Var, Op0W, {});
+        auto Op1 = IC.getInst(Inst::Var, Op1W, {});
+        auto Op2 = IC.getInst(Inst::Var, Op2W, {});
+        auto I = IC.getInst(K, WIDTH, {Op0, Op1, Op2});
+        std::unordered_map<Inst *, llvm::KnownBits> C{{Op0, x}, {Op1, y}, {Op2, z}};
```

Comment:
it doesn't really matter in this kind of testing code, so you don't need to fix, but creating a map in a tight loop is pretty inefficient, something like a SmallMapVector would be better since it won't hit the allocator